medusa/helpers.py

```diff
@@ -860,7 +860,9 @@ def check_url(url):
 
 def anon_url(*url):
     """Return a URL string consisting of the Anonymous redirect URL and an arbitrary number of values appended."""
-    return '' if None in url else '%s%s' % (app.ANON_REDIRECT, ''.join(str(s) for s in url))
+    # normalize to byte
+    url = [u.encode('utf-8') if isinstance(u, unicode) else str(u) for u in url]
```

Comment:
It seems ok, but I would use six instead of unicode or str:
`url = [u.encode('utf-8') if isinstance(u, text_type) else binary_type(u) for u in url]`